server.go

```diff
@@ -403,38 +360,43 @@ func (a *App) handle_scrubbing() error {
 	return nil
 }
 
-func (a *App) run_server() {
+func (a *App) runServer(ctx context.Context) {
 	log.Info().Int("twin", a.twin).Msg("initializing agent server")
 
 	for {
-		res, err := a.redis.BLPop(a.ctx, 1000000000, "msgbus.system.local", "msgbus.system.remote", "msgbus.system.reply").Result()
+		res, err := a.redis.BLPop(a.ctx, time.Second, "msgbus.system.local", "msgbus.system.remote", "msgbus.system.reply")
+
+		if err == nil {
+			log.Debug().Str("queue", res[0]).Str("message", res[1]).Msg("found a redis payload")
+		}
 
 		if err == redis.Nil {
-			a.handle_retry()
-			a.handle_scrubbing()
+			a.handleRetry()
```

Comment:
As discussed before, you should **NEVER** ignore errors, you still can only log them. but never ignored.